### ISSUE TITLE
Setting ClientDatabasePrefix via builder causes crash if configuration is present

### ISSUE THREAD

**Issue by mslagle-gd** [awslabs/aws-mobile-appsync-sdk-android#306]
**Describe the bug**
Setting a `clientDatabasePrefix` programmatically causes a crash if there is a configuration

**To Reproduce**
```
AWSAppSyncClient.builder()
                .context(context)
                .awsConfiguration(appSyncConfig)
                .region(fromName(awsRegion))
                .serverUrl(awsEndpoint)
                .credentialsProvider(credentialsProvider)
                .clientDatabasePrefix("SomePrefix")
                .useClientDatabasePrefix(true)
                .build()
```

**Expected behavior**
It sets the `clientDatabasePrefix` on the builder, that prefix is used, and the library doesn't throw an error.

**Environment(please complete the following information):**
 - AppSync SDK Version: 3.1.0

**Device Information (please complete the following information):**
 - Device: Pixel 3, Simulator
 - Android Version: 10

**Additional context**
The issue is [here](https://github.com/awslabs/aws-mobile-appsync-sdk-android/blob/main/aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java#L561). If you set the `clientDatabasePrefix` using the `Builder.clientDatabasePrefix()` method, but have a config, then it overrides your previously set value with `null` instead of checking if one is already set programmatically.

```
Caused by: java.lang.RuntimeException: Please check the AppSync configuration in awsconfiguration.json.
        at com.amazonaws.mobileconnectors.appsync.AWSAppSyncClient$Builder.build(AWSAppSyncClient.java:589)
        at com.godaddy.leka.appsync.AppSyncRegistrationCoordinator$registerAppSyncClient$$inlined$registerWithOverride$1$1.invoke(DependencyService.kt:361)
        at com.godaddy.leka.appsync.AppSyncRegistrationCoordinator$registerAppSyncClient$$inlined$registerWithOverride$1$1.invoke(DependencyService.kt:115)
        at org.koin.core.instance.DefinitionInstance.create(DefinitionInstance.kt:54)
        	... 44 more
     Caused by: java.lang.RuntimeException: ClientDatabasePrefix is not present in AppSync configuration in awsconfiguration.json however .useClientDatabasePrefix(true) is passed in.
        at com.amazonaws.mobileconnectors.appsync.AWSAppSyncClient$Builder.build(AWSAppSyncClient.java:564)
```

**Comment by jamesonwilliams** [comment]
@rjuliano Does #308 resolve #306? Do we just need to release it?

**Comment by rjuliano** [comment]
> @rjuliano Does #308 resolve #306? Do we just need to release it?

That's correct.

**Comment by jamesonwilliams** [comment]
Released in 3.1.1.
